packages/web/src/lib/web-mode.ts

```diff
@@ -184,62 +185,49 @@ export class WebMode {
 
 	/**
 	 * Appends signature to a multisig transaction using algosigner
-	 * @param txns array of multisignature transaction with at least one signature
+	 * @param txn Multisignature Encoded Transaction  
 	 * @param signers a subset of addresses to sign the transaction
-	 * return an object containing a blob attribute encoded in base64
+	 * return an object containing a blob key encoded in base64
 	 */
 	async appendSignMultisigTransaction(
-		txns: WalletTransaction[],
-		signers: string[]
+		txn: EncodedSignedTransaction,
+		signers?: string[]
 	): Promise<JsonPayload> {
 		try {
-			const result: JsonPayload = {};
-			for (let i = 0; i < txns.length; ++i) {
-				const txn = txns[i];
-				const partialTxn = algosdk.decodeObj(
-					this.algoSigner.encoding.base64ToMsgpack(txn.txn)
-				) as EncodedSignedTransaction;
-				if (partialTxn.txn === undefined || partialTxn.msig === undefined) {
-					throw new Error(
-						"Input transaction must be multisigature transaction signed with at least 1 signature"
-					);
-				}
-				const txnToBeSign = algosdk.Transaction.from_obj_for_encoding(partialTxn.txn);
-				const txnToBeSign_Uint8Array = algosdk.encodeObj(txnToBeSign.get_obj_for_encoding());
-				const txnToBeSign_Base64 =
-					this.algoSigner.encoding.msgpackToBase64(txnToBeSign_Uint8Array);
-
-				const mparams = partialTxn.msig as algosdk.EncodedMultisig;
-				const addrs = mparams.subsig.map((signData) => {
-					return algosdk.encodeAddress(signData.pk);
-				});
+			const encodedTxn = this.algoSigner.encoding.msgpackToBase64(encodeObj(txn.txn))
```

Comment:
Shouldn't we need to check whether the txn input should be a multisig transaction first?

Reply:
the input parameter is already updated from `WalletTransaction` to `EncodedSignedTransaction`

Reply:
However, in type [EncodedSignedTransaction](https://algorand.github.io/js-algorand-sdk/interfaces/EncodedSignedTransaction.html), `msig` is an optional parameter so you still need to check it

Reply:
oh okay, will update